train-tracker/src/Search.js

```diff
@@ -13,17 +13,33 @@ function Search({searchFun, routes, stations}){
     const [upcoming, setUpcoming] = useState(false);
     const [fromStation, setFromStation] = useState("");
     const [toStation, setToStation] = useState("");
+    const [favoriteOptions, addToFavList] = useState(populateFavDrop)
 
     function handleNumber(e){ setSelectedNumber(e.target.value); }
     function handleRoute(e){ setSelectedRoute(e.target.value); }
     function handleStation(e){ setSelectedStation(e.target.value); }
     function handleUpcoming(e){ setUpcoming(e.target.checked); }
     function handleFromStation(e){ setFromStation(e.target.value); }
     function handleToStation(e){ setToStation(e.target.value); }
-    function setToFavorites(){
-        setToCache(selectedRoute,selectedNumber, selectedStation);
+    
+    function handleFavoriteSelection(e){
+        addToFavList(populateFavDrop())
+        e.preventDefault();
+        searchFun("", e.target.value, "", "", "", "")
```

Comment:
this seems to activate whenever the box is clicked, for example erasing your current search rather than searching once you make a selection. Is that possible to change? I'm not sure if the limits of dropdowns so this might not be possible